src/ansys/geometry/core/geometry/box_uv.py

```diff
@@ -0,0 +1,184 @@
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Provides the ``BoxUV`` class."""
+from enum import Enum
+
+from ansys.geometry.core.geometry.parameterization import Interval, ParamUV
+from ansys.geometry.core.typing import Real
+
+
+class LocationUV(Enum):
+    """Provides the ``LocationUV`` class to indicate locations for BoxUV."""
+
+    TopLeft = 1
+    TopCenter = 2
+    TopRight = 3
+    BottomLeft = 4
+    BottomCenter = 5
+    BottomRight = 6
+    LeftCenter = 7
+    RightCenter = 8
+    Center = 9
+
+
+class BoxUV:
+    """BoxUV class."""
+
+    def __init__(self, range_u: Interval = None, range_v: Interval = None) -> None:
+        """Root constructor for BoxUV."""
+        if range_u is not None:
+            self._interval_u = range_u
+        if range_v is not None:
+            self._interval_v = range_v
+
+    @classmethod
+    def from_param(cls, param: ParamUV):
+        """Secondary constructor for BoxUV using a ParamUV object type."""
+        return cls(Interval(param.u, param.u), Interval(param.v, param.v))
+
+    @classmethod
+    def from_two_params(cls, param1: ParamUV, param2: ParamUV):
+        """Secondary constructor for BoxUV using two ParamUV object types."""
+        return cls(
+            Interval(min(param1.u, param2.u), max(param1.u, param2.u)),
+            Interval(min(param1.v, param2.v), max(param1.v, param2.v)),
+        )
+
+    @property
+    def interval_u(self) -> Interval:
+        """Return the u interval."""
+        return self._interval_u
+
+    @property
+    def interval_v(self) -> Interval:
+        """Return the v interval."""
+        return self._interval_v
+
+    def __eq__(self, other: object) -> bool:
+        """Check whether two BoxUV instances are equal."""
+        if not isinstance(other, BoxUV):
+            # don't attempt to compare against unrelated types
+            return NotImplemented
+        return self.interval_u.__eq__(other.interval_u) and self.interval_v.__eq__(other.interval_v)
+
+    def __ne__(self, other: object) -> bool:
+        """Check whether two BoxUV instances are not equal."""
+        if not isinstance(other, BoxUV):
+            # don't attempt to compare against unrelated types
+            return NotImplemented
+        return not self.interval_u.__eq__(other.interval_u) or not self.interval_v.__eq__(
+            other.interval_v
+        )
+
+    def is_empty(self):
+        """Return whether this BoxUV is empty."""
```

Comment:
```suggestion
        """Check if this BoxUV is empty."""
```
Again, descriptions should use plain English. If that isn't possible, then the non-English word should be tagged as a code entity.